src/PanoImageRenderer/ImageLoader.js

```diff
@@ -1,72 +1,86 @@
-		this.image = null;
+		this._image = null;
+		this._loadStatus = STATUS.NONE;
 
-		image && this.setImage(image);
+		image && this.set(image);
 	}
 
 	get() {
 		return new Promise((res, rej) => {
-			if (!this.image) {
-				rej("image is not defiend");
-			} else if (this.image.complete) {
-				res(this.image);
-			} else {
-				ImageLoader._once(this.image, "load", () => {
-					res(this.image);
+			if (!this._image) {
+				rej("ImageLoader: image is not defiend");
+			} else if (this._loadStatus === STATUS.LOADED || ImageLoader._isMaybeLoaded(this._image)) {
+				/* Check isMaybeLoaded() first because there may have posibilities that image already loaded before get is called. for example calling get on external image onload callback.*/
+				res(this._image);
+			} else if (this._loadStatus === STATUS.LOADING) {
+				ImageLoader._once(this._image, "load", () => {
+					this._loadStatus = STATUS.LOADED;
+					res(this._image);
 				});
-				ImageLoader._once(this.image, "error", () => {
-					rej("failed to load images.");
+				ImageLoader._once(this._image, "error", () => {
+					this._loadStatus = STATUS.ERROR;
+					rej("ImageLoader: failed to load images.");
 				});
+			} else {
+				rej("ImageLoader: failed to load images");
 			}
 		});
 	}
 
-	setImage(image) { // img element or img url
+	/**
+	 * @param image img element or img url
+	 */
+	set(image) {
 		if (typeof image === "string") {
-			this.image = new Image();
-			this.image.src = image;
-		} else if (typeof image === "object") { // img element 나 image object 이어야 함
-			this.image = image;
+			this._image = new Image();
+			this._loadStatus = STATUS.LOADING;
+			this._image.onload = () => {
+				this._loadStatus = STATUS.LOADED;
+			};
+			this._image.onerror = () => {
+				this._loadStatus = STATUS.ERROR;
+			};
+			this._image.src = image;
+		} else if (typeof image === "object") {
+			if (ImageLoader._isMaybeLoaded(image)) {
+				this._loadStatus = STATUS.LOADED;
+			} else {
+				this._loadStatus = STATUS.LOADING;
+			}
+			this._image = image;
 		}
 
 		// promise for image
 		return this.get();
 	}
 
+	static _isMaybeLoaded(image) {
+		return image && image.naturalWidth !== 0;
+	}
+
 	static _once(target, type, listener) {
 		target.addEventListener(type, function fn(event) {
 			target.removeEventListener(type, fn);
 			listener(event);
 		});
-		// target.addEventListener(type, listener);
-	}
-
-	isImageLoaded() {
-		if (!this.image) {
-			return false;
-		}
-
-		return !!this.image.src && !!this.image.complete;
-	}
-
-	/**
-	 * TODO: 이 기능이 정말로 필요한가?
-	 */
-	cancelLoadImage() {
-		if (!!this.image && !this.isImageLoaded()) {
-			this.image.src = "";
-		}
-	}
-
-	loadImage() {
-		if (this._imageURL && !this.isImageLoaded()) {
-			this.image.setAttribute("crossorigin", "anonymous");
-			this.image.src = this._imageURL;
-		}
 	}
 
 	destroy() {
-		this.image = null;
+		/**
+		 * Init event handlers
+		 */
+		this._image.onload = null;
+		this._image.onerror = null;
```

Comment:
@happyhj Initialize event handler for preventing event from being fired after destroy